### PR TITLE
merge_request_5_feature_expose_endpoints_and_crud_actions(fire_station)

### PR DESCRIPTION
**Pull request by G-jonathan**
Merge request which corresponds to the feature: _Expose the following endpoint : http://localhost:8080/firestation_
This endpoint will allow to perform the following actions: Create a new fire station mapping, update a fire station number address and delete a mapping of station or an address.

The service package includes the interfaces which are then implemented in the classes of the service/impl package.

### CODE REVIEW

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/SafetyNetAlertsApplication.java [hunk 1]
```diff
@@ -17,13 +17,19 @@ public static void main(String[] args) {
...
+	CommandLineRunner runnerReadJsonWithObjectMapper(JsonDataService jsonDataService) {
 		return args -> {
 			ObjectMapper objectMapper = new ObjectMapper();
 			JsonFileData jsonFileData = objectMapper.readValue(new File("src/main/resources/json/data.json"), JsonFileData.class);
 			jsonDataService.savePersons(jsonFileData.getPersons());
-			jsonDataService.saveFirestations(jsonFileData.getFirestations());
+			jsonDataService.saveFireStations(jsonFileData.getFirestations());
 			jsonDataService.saveMedicalRecords(jsonFileData.getMedicalrecords());
+
+			/*
```
Comment (Asavarkhul) [inline src/main/java/com/openClassroomsProject/SafetyNetAlerts/SafetyNetAlertsApplication.java:28]: Dead code

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/JsonDataService.java [hunk 2]
```diff
@@ -18,23 +18,23 @@
     @Autowired
     private MedicalRecordRepository medicalrecordRepository;
     @Autowired
-    private FirestationRepository firestationRepository;
+    private FireStationRepository firestationRepository;
```
Comment (Asavarkhul) [inline src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/JsonDataService.java:21]: Should be an interface

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/FireStationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class FireStationServiceImpl implements FireStationService {
+
+    @Autowired
+    private FireStationRepository fireStationRepository;
```
Comment (Asavarkhul) [inline src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/FireStationServiceImpl.java:15]: Should be an interface

File: src/test/java/com/openClassroomsProject/SafetyNetAlerts/Controller/FireStationControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import org.springframework.test.web.servlet.MockMvc;
+
+@WebMvcTest(controllers = FireStationController.class)
+public class FireStationControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+    @MockBean
+    private JsonDataService jsonDataService;
+    @MockBean
+    private FireStationServiceImpl fireStationService;
```
Comment (Asavarkhul) [inline src/test/java/com/openClassroomsProject/SafetyNetAlerts/Controller/FireStationControllerTest.java:22]: Should be an interface

**Review by Asavarkhul — CHANGES_REQUESTED** [review]

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/SafetyNetAlertsApplication.java (hunk 1 above)
Comment (G-jonathan) [reply src/main/java/com/openClassroomsProject/SafetyNetAlerts/SafetyNetAlertsApplication.java]: Done

File: src/test/java/com/openClassroomsProject/SafetyNetAlerts/Controller/FireStationControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @Test
+    public void testDeleteMappingOfAnAddress() throws Exception {
+        mockMvc.perform(delete("/fireStation/address/{address}"))
+                .andExpect(status().isOk());
+    }
+}
+
+
```
[the remark this answers is not in the document]
Comment (G-jonathan) [reply src/test/java/com/openClassroomsProject/SafetyNetAlerts/Controller/FireStationControllerTest.java]: Done

File: src/test/java/com/openClassroomsProject/SafetyNetAlerts/Controller/FireStationControllerTest.java (hunk 4 above)
Comment (G-jonathan) [reply src/test/java/com/openClassroomsProject/SafetyNetAlerts/Controller/FireStationControllerTest.java]: Done

File: src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/FireStationServiceImpl.java (hunk 3 above)
Comment (G-jonathan) [reply src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/impl/FireStationServiceImpl.java:15]: FirestationRepository is an interface ^^

**Pull request merged**